core/shared/src/main/scala/zio/Cause.scala

```diff
@@ -323,13 +353,25 @@ object Cause extends Serializable {
     override final def hashCode: Int = cause.hashCode()
     override final def equals(obj: Any): Boolean = obj match {
       case traced: Traced[_] => cause == traced.cause
+      case meta: Meta[_]     => cause == meta.cause
+      case _                 => cause == obj
+    }
+  }
+
+  // Meta is excluded completely from equals & hashCode
+  final case class Meta[E](cause: Cause[E], data: Data) extends Cause[E] {
```

Comment:
It may be time to make all these case classes / objects private so we have more flexibility to change `Cause` during the 1.x lifecycle. What do you think?

A basic fold method that hides `Meta` could be used to allow people to traverse.

Reply:
I think that makes sense. So then `fold` would go to needing to take a parameter for each constructor of `Cause`, because we couldn't allow people to pattern match against the `Cause` constructors, right?